### PR TITLE
More slime extract uses

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[content]

## What this does
cleans up the extract adding system to allow slime extracts to be applied to any object and for all types, regardless of it having a use or not
lets silver slimes cook "perfect" forms of some recipes like plump helmet pie every time
adds more chem dispenser options for adding black and pyrite slimes; banana juice and colorful reagent for pyrite on bar dispensers and chemical ones respectively, and blood for space breweries and tricordrazine for other bar dispensers
lets pyrite slimes multiply the results of certain ores mined, with values of 2 for gold silver and uranium, 4 for diamond and 8 for phazon bananium telecrystal and mythril. it also boosts the odds of finding dusty crates within.
expands the system for slime reagent refills on spray bottles, allowing green slimes to refill luminol, red slimes to refill plantbgone and insecticide
lets slime extracts run a small amount of power automatically on port gens, with darkpurple needed for pacman, additional metal for superpacman and additional adamantine for mrspacman
lets red slime extracts refuel bunsen burners with a version of glycerol with no bad side effects

## Why it's good
makes them more useful

## How it was tested
adding the relevant extracts and doing the following:
cooking relevant recipes a few times in the microwave and pan
running port generators
mining ores
using reagent dispensers
using the bunsen burner
using the relevant spray bottles

## Changelog
:cl:
 * rscadd: Silver slime extracts now allow for the creation of cooking recipes in their random probability "perfect" forms, such as frosted donuts.
 * rscadd: Red slime extracts now refill bunsen burners with glycerol and remove its downsides, while refilling itself.
 * rscadd: Pyrite slime extracts now boost the output of mining tools depending on ore, with diamond multiplying by 4 and gold/silver/uranium by 2, and dusty crates becoming more common finds.
 * rscadd: Insecticide and plantbgone sprays can now be refilled with red slime extracts, luminol and fakecreep with green and pyrite respectively.
 * rscadd: Port generators can now accept slime extracts to run a small amount of extra automatic output. PACMAN requires dark purple, SUPERPACMAN requires dark purple and metal, MRSPACMAN requires dark purple, metal and adamantine, BABYPACMAN requires dark purple, metal, adamantine and pyrite, and PROFESSORPACMAN requires dark purple, metal, adamantine, pyrite and black.
 * rscadd: Bar dispensers can now dispense banana juice and chemical dispensers can now dispense colorful reagent by adding pyrite slime extract to them. Bar dispensers now have different reagents dispensed to delightful syrup when black slime extract is applied; blood for space breweries and tricordrazine for everything else.

### CODE REVIEW

**Comment by DeityLink** [comment]
I'd argue that Xenobio needs less rewards and more of a job rework/QoL pass if you want players to actually play it. It's already got plenty of great rewards, but managing slimes is tedious and repetitive. It was fun to learn the first time, but now I'd rather play any other job with less epic rewards but more engaging gameplay.

**Comment by SECBATON-GRIFFON** [comment]
> I'd argue that Xenobio needs less rewards and more of a job rework/QoL pass if you want players to actually play it. It's already got plenty of great rewards, but managing slimes is tedious and repetitive. It was fun to learn the first time, but now I'd rather play any other job with less epic rewards but more engaging gameplay.

i wouldnt mind doing some next. care to give ideas?

**Comment by DeityLink** [comment]
A long time idea I've had in the back of my mind is xenobio HUD glasses that inform you whether the slimes consider you a friend or not, how hungry/electricity-charged they are/how closed to splitting.

Additionally Fire Extinguishers are long overdue for a proper fix of their spread pattern, it often makes no sense and the steam around the tile that you click at.

Something like the gun in Slime Rancher that can vacuum in live slimes, store and release them, would make moving slimes between pens infinitely less of a hassle (with a risk of it glomping you instead if you try to suck it in while it doesn't consider you a friend yet). Players could also use it to shoot slimes at people, automatically glomping them.

Bio suits ACTUALLY offering some protection from slimes so you don't take clone damage right away.

These are just a few ideas.

File: code/modules/mining/minerals.dm [hunk 1]
```diff
@@ -159,6 +168,7 @@ var/list/name_to_mineral
 	name = "Telecrystal"
 	result_amount = 1
 	spread_chance = 5
+	fortune_multiplier = 8
```
Comment (jwhitak) [inline code/modules/mining/minerals.dm:171]: This was rejected by the administration, please remove the modifier for telecrystals. (You can insert them into an uplink for more TC)

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -449,6 +454,7 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 	icon_state = "brewer"
 	pass_flags = PASSTABLE
 	required_quirk = MODULE_CAN_HANDLE_FOOD
+	slime_reagents = list("black" = BLOOD, "pyrite" = BANANA)
```
Comment (jwhitak) [inline code/modules/reagents/machinery/chem_dispenser.dm:457]: I personally find it a little unusual that this makes blood intstead of tricord here. How do others think?

**Comment by angryturnip** [comment]
I don't think TC should have an 8x multiplier, frankly. But if that gets removed I think the rest is good

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -972,6 +984,7 @@ var/list/icon_state_to_appearance = list()
 	name = "Telecrystal deposit"
 	icon_state = "rock_Telecrystal"
 	mineral = new /mineral/telecrystal
+	fortune_multiplier = 8
```
Comment (jwhitak) [inline code/modules/mining/mine_turfs.dm:987]: See other comment

**Review by jwhitak — CHANGES_REQUESTED** [review]

File: code/modules/mining/minerals.dm (hunk 1 above)
Comment (SECBATON-GRIFFON) [reply code/modules/mining/minerals.dm:171]: the ore actually produces bluespace crystals, plus the kind that get inserted into uplinks are called *refined* telecrystals, and are of type /obj/item/stack/telecrystal (not even /obj/item/stack/sheet/mineral/telecrystal) and have zero connection to this

File: code/modules/reagents/machinery/chem_dispenser.dm (hunk 2 above)
Comment (SECBATON-GRIFFON) [reply code/modules/reagents/machinery/chem_dispenser.dm:457]: really its for easier lifeline

File: code/modules/mining/mine_turfs.dm (hunk 3 above)
Comment (SECBATON-GRIFFON) [reply code/modules/mining/mine_turfs.dm:987]: seen

**Comment by SECBATON-GRIFFON** [comment]
> I don't think TC should have an 8x multiplier, frankly. But if that gets removed I think the rest is good

the TC mineral only produces bluespace crystals and isn't used in uplinks in any form, see my comment under the requested change

File: code/modules/mining/minerals.dm (hunk 1 above)
Comment (jwhitak) [reply code/modules/mining/minerals.dm:171]: I swear I checked it before posting this!!! The sheet/mineral has trolled me again...

**Comment by SECBATON-GRIFFON** [comment]
> A long time idea I've had in the back of my mind is xenobio HUD glasses that inform you whether the slimes consider you a friend or not, how hungry/electricity-charged they are/how closed to splitting.

late but you can get the electricity thing from examining them already.

**Pull request merged**